include/mptcpd/listener_manager.h

```diff
@@ -31,16 +31,26 @@ struct mptcpd_lm;
  * @param[in] lm The mptcpd address listener manager object.
  * @param[in] sa The MPTCP local address.
  *
- * @return @c true on success, and @c false on failure.
+ * @return Non-zero port in network byte order to which the listener
+ *         was bound, and zero on failure.  An ephemeral port will be
+ *         returned if the port in the local address @a sa is zero.
+ *         The @c sockaddr passed to subsequent calls to
+ *         @c mptcpd_lm_close() should take this into account since
+ *         the mptcpd listener manager will only keep track of
+ *         addresses with non-zero ports, including addresses with
+ *         ephemeral ports.
```

Comment:
From the code below, it looks like returning `0` indicates failure. Should mention that in the comment here.

Reply:
I had thought of returning an `int` (`errno`) instead of the port, and returning the port through a pointer parameter.  Do you have a preference?

Reply:
`0` indicating failure is mentioned in the first sentence.  I guess it's buried in a bunch of text.  I'll try to make it clearer.

Reply:
Sorry, it is of course right there... Not sure how/why my eyes skipped over that.

During the meeting this morning it sounded like you were considering the sockaddr `sin_port`/`sin6_port` for propagating the port number?

Reply:
Yes.  Commit 47bafca provides the port propagation we discussed.